core/htmlparser/filter.js

```diff
@@ -402,6 +402,140 @@
 } )();
 
 /**
+ * Abstract class describing the definition of {@link CKEDITOR.htmlParser.filter} rules.
+ *
+ * This class illustrates the properties that developers can use to define and create filter rules.
+ *
+ * A filter rules definition object represents rules as a set of properties with callback functions
+ * to be applied for transforming and filtering content upon data processing.
+ *
+ * Filter rules definition can be also added with {@link CKEDITOR.htmlParser.filter.addrules}.
+ *
  * @class CKEDITOR.htmlParser.filterRulesDefinition
  * @abstract
  */
+
+/**
+ * @property {String[][]/RegExp[][]} elementNames An array of rules for data processing element names.
+ * Rule is defined as an array with two items [ string/regex, string ].
+ * Every matching string from first item will be converted into second.
+ *
+ * Examples:
+ *
+ * ```javascript
+ * elementNames: [
+ * 		[ /^div$/, 'p' ], // Converts 'div' into 'p'.
+ * 		[ /^cke:?/, '' ] // Removes 'cke:' prefixes.
+ * ]
+ * ```
+ *
+ * Incorrect:
```

Comment:
Examples already describe correct rule regex, not sure if we need this section. It seems obvious to me that if you are matching string by regex, too small specificity will give you invalid results. WDYT?